pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -254,20 +254,35 @@ public SuccessResponse deleteTable(
 
     List<String> tablesDeleted = new LinkedList<>();
     try {
-      if (tableType != TableType.REALTIME && !TableNameBuilder.REALTIME.tableHasTypeSuffix(tableName)) {
+      if (verifyTableType(tableName, tableType, tableType.OFFLINE) && _pinotHelixResourceManager
+          .hasOfflineTable(tableName)) {
         _pinotHelixResourceManager.deleteOfflineTable(tableName);
         tablesDeleted.add(TableNameBuilder.OFFLINE.tableNameWithType(tableName));
       }
-      if (tableType != TableType.OFFLINE && !TableNameBuilder.OFFLINE.tableHasTypeSuffix(tableName)) {
+      if (verifyTableType(tableName, tableType, tableType.REALTIME) && _pinotHelixResourceManager
+          .hasRealtimeTable(tableName)) {
         _pinotHelixResourceManager.deleteRealtimeTable(tableName);
         tablesDeleted.add(TableNameBuilder.REALTIME.tableNameWithType(tableName));
       }
+      if (tablesDeleted.size() == 0) {
+        throw new ControllerApplicationException(LOGGER, "Table '" + tableName + "' does not exist",
+            Response.Status.NOT_FOUND);
+      }
       return new SuccessResponse("Tables: " + tablesDeleted + " deleted");
     } catch (Exception e) {
-      throw new ControllerApplicationException(LOGGER, e.getMessage(), Response.Status.INTERNAL_SERVER_ERROR, e);
+      throw new ControllerApplicationException(LOGGER, e.getMessage(),
+          Response.Status.INTERNAL_SERVER_ERROR, e);
     }
   }
 
+  // Return true iff the table is of the expectedType based on the given tableName and tableType.
+  private boolean verifyTableType(String tableName, TableType tableType, TableType expectedType) {
+    if (tableType == null)
+      return TableNameBuilder.getTableTypeFromTableName(tableName) == expectedType ||
```

Comment:
Please add braces for statement in the if clause, and also cache the tableType into a local variable to save one string comparison.